pkg/status/certificate/certificate.go

```diff
@@ -308,9 +308,8 @@ func findMatchingCR(cmClient cmclient.Interface, ctx context.Context, crt *cmapi
 		nextRevision = *crt.Status.Revision + 1
 	}
 	for _, req := range reqs.Items {
-		// #nosec G601 -- False positive. See https://github.com/golang/go/discussions/56010
-		if predicate.CertificateRequestRevision(nextRevision)(&req) &&
-			predicate.ResourceOwnedBy(crt)(&req) {
+		if predicate.CertificateRequestRevision(nextRevision)(&req) && /* #nosec G601 -- Pointer does not outlive function scope */
+			predicate.ResourceOwnedBy(crt)(&req) /* #nosec G601 -- Pointer does not outlive function scope */ {
```

Comment:
Mixing comments and logic makes it hard to spot the `&&` and `{`. Why not have line comments instead?

Reply:
These `#nosec` require the comment to be placed very close to the code, I remember that making this a line comment did not work.
see https://github.com/securego/gosec/issues/278#issuecomment-745209803

Reply:
Ah..... a limitation due to how `// #nosec` is scoped when walking the AST, apparently. It's a shame that a linter forces us to write unreadable code.

Reply:
Indeed, I forgot to initially explain in the PR description that this was not MY choice 🥲.